src/main/java/org/broadinstitute/consent/http/service/DataAccessRequestService.java

```diff
@@ -363,6 +364,11 @@ public DataAccessRequest cancelDataAccessRequest(String referenceId) {
         if (Objects.isNull(dar)) {
             throw new NotFoundException("Unable to find Data Access Request with the provided id: " + referenceId);
         }
+        List<Election> elections = electionDAO.findElectionsByReferenceId(referenceId);
+        if (!elections.isEmpty()) {
+            //Is this the right exception to use?
+            throw new IllegalArgumentException("Cancelling this DAR is not allowed");
```

Comment:
I think `UnsupportedOperationException` is more appropriate here. The reference id is technically a valid argument, but we are choosing not to perform the requested operation due to business policy. You could even make an argument that this is a 409 conflict exception, but I think that's a bit of a stretch.